From 47e255a7ef896792c2abab02f14dfb44f3cfd1f5 Mon Sep 17 00:00:00 2001
From: Xenia Ragiadakou <xenia.ragiadakou@amd.com>
Date: Thu, 14 Sep 2023 21:26:48 +0300
Subject: [PATCH 207/221] pvh dom0: fix MSI-x initialization for passed-through
 devices

As part of XSA-337, reading BARs is not allowed once the device is
assigned to a guest. The reading of BARs is performed earlier by Dom0
in the pcistub probe. For this, Dom0 needs to support the hypercalls
PHYSDEVOP_prepare_msix and PHYSDEVOP_release_msix. As of now, these
physdevop commands are available only to pv guests.
Enable them also for hvm hardware domains to fix MSI-x initialization
when Dom0 is pvh.

Signed-off-by: Xenia Ragiadakou <xenia.ragiadakou@amd.com>
Change-Id: If2e7272b5abb044fd5582be7e7c0e60b4ff8e635
Signed-off-by: Ramesh Babu Nalluri <rameshbabu.nalluri@amd.com>
---
 xen/arch/x86/hvm/hypercall.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/xen/arch/x86/hvm/hypercall.c b/xen/arch/x86/hvm/hypercall.c
index fce786618c..786f2a5000 100644
--- a/xen/arch/x86/hvm/hypercall.c
+++ b/xen/arch/x86/hvm/hypercall.c
@@ -96,6 +96,8 @@ long hvm_physdev_op(int cmd, XEN_GUEST_HANDLE_PARAM(void) arg)
     case PHYSDEVOP_pci_device_add:
     case PHYSDEVOP_pci_device_remove:
     case PHYSDEVOP_dbgp_op:
+    case PHYSDEVOP_prepare_msix:
+    case PHYSDEVOP_release_msix:
         if ( !is_hardware_domain(currd) )
             return -ENOSYS;
         break;
-- 
2.17.1

